### PR TITLE
[ADD] account_payment_return_import: New module base

### PR DESCRIPTION
**Pull request by carlosdauden**

### CODE REVIEW

File: account_payment_return_import/README.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
```
Comment (pedrobaeza) [inline account_payment_return_import/README.rst]: Put target line

File: account_payment_return_import/README.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Account Payment Return Import
```
Comment (pedrobaeza) [inline account_payment_return_import/README.rst:6]: Line above

File: account_payment_return_import/README.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+=============================
+
+This module add a generic wizard to import payment return file formats.
+Is only the base to be extended by another modules.
+
+Known issues / Roadmap
+======================
+
+* Better...
```
Comment (pedrobaeza) [inline account_payment_return_import/README.rst]: ?

File: account_payment_return_import/README.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Account Payment Return Import
+=============================
+
+This module add a generic wizard to import payment return file formats.
```
Comment (pedrobaeza) [inline account_payment_return_import/README.rst]: s/add/adds

Comment (pedrobaeza) [inline account_payment_return_import/README.rst]: s/wizard/wizard + methods

File: account_payment_return_import/i18n/es.po [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+#: view:payment.return.import:account_payment_return_import.account_payment_return_import_view
+msgid "_Import"
+msgstr "Importado"
+
+#. module: account_payment_return_import
+#: view:payment.return.import:account_payment_return_import.account_payment_return_import_view
+msgid "or"
+msgstr "o"
+
+#~ msgid "Bank Statement File"
```
Comment (pedrobaeza) [inline account_payment_return_import/i18n/es.po]: Remove commented lines

File: account_payment_return_import/README.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+Is only the base to be extended by another modules.
```
Comment (pedrobaeza) [inline account_payment_return_import/README.rst]: It's

Comment (pedrobaeza) [inline account_payment_return_import/README.rst]: Include a note indicating that zip files with return files are also supported.

File: account_payment_return_import/models/account_payment_return_import.py [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+        help='Get you bank payment returns in electronic format from your '
+             'bank and select them here.')
+
+    @api.multi
+    def import_file(self):
+        """Process the file chosen in the wizard, create bank payment return(s)
+        and go to reconciliation."""
+        self.ensure_one()
+        data_file = base64.b64decode(self.data_file)
+        # pylint: disable=protected-access
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: These checks are not in OCA Travis, so you don't need to disable it.

File: account_payment_return_import/models/account_payment_return_import.py [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+        payment_return_ids, notifications = self.with_context(
+            active_id=self.id  # pylint: disable=no-member
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: The same for pylint

File: account_payment_return_import/models/account_payment_return_import.py [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+        )._import_file(data_file)
+        # dispatch to reconciliation interface
+        result = self.env.ref(
+            'account_payment_return.payment_return_action').read()[0]
+        if len(payment_return_ids) != 1:
+            result['domain'] = "[('id', 'in', %s)]" % payment_return_ids
+        else:
+            res = self.env.ref(
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: Use self.ref directly to get the id

File: account_payment_return_import/models/account_payment_return_import.py [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+        # Check raw data:
+        self._check_parsed_data(payment_returns)
+        # Import all payment returns:
+        for payret_vals in payment_returns:
+            (payment_return_id, new_notifications) = (
+                self._import_payment_return(payret_vals))
+            if payment_return_id:
+                payment_return_ids.append(payment_return_id)
+            notifications.extend(new_notifications)
+        if len(payment_return_ids) == 0:
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: if not payment_returns:

File: account_payment_return_import/models/account_payment_return_import.py [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+            # returns. Actually we don't care wether all the files have the
+            # same format.
+            payment_returns += self._parse_file(import_file)
+        return payment_returns
+
+    @api.model
+    def _import_file(self, data_file):
+        """ Create bank payment return(s) from file."""
+        # The appropriate implementation module returns the required data
+        payment_return_ids = []
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: Call it payment_returns and make it a recordset

File: account_payment_return_import/models/account_payment_return_import.py [hunk 12]
```diff
@@ -0,0 +1,303 @@
...
+            raise UserError(_('Can not determine journal for import.'))
+        # Prepare payment return data to be used for payment returns creation
+        payret_vals = self._complete_payment_return(
+            payret_vals, journal_id, account_number)
+        # Create the bank payret_vals
+        return self._create_payment_return(payret_vals)
+
+    @api.model
+    def _parse_file(self, data_file):
+        # pylint: disable=no-self-use
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: Remove these disable. They are not check in Travis

File: account_payment_return_import/models/account_payment_return_import.py [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+        raise UserError(_(
+            'Could not make sense of the given file.\n'
+            'Did you install the module to support this type of file?'
+        ))
+
+    @api.model
+    def _check_parsed_data(self, payment_returns):
+        # pylint: disable=no-self-use
+        """ Basic and structural verifications """
+        if len(payment_returns) == 0:
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: if not payment_returns:

File: account_payment_return_import/models/account_payment_return_import.py [hunk 14]
```diff
@@ -0,0 +1,303 @@
...
+            raise UserError(_(
+                'This file doesn\'t contain any payment return.'))
+        for payret_vals in payment_returns:
+            if 'transactions' in payret_vals and payret_vals['transactions']:
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: Use 
if not payret_vals.get('transactions'):
    raise ...

File: account_payment_return_import/models/account_payment_return_import.py [hunk 15]
```diff
@@ -0,0 +1,303 @@
...
+        # If we get here, no transaction was found:
+        raise UserError(_('This file doesn\'t contain any transaction.'))
+
+    @api.model
+    def _find_bank_account_id(self, account_number):
+        """ Get res.partner.bank ID """
+        bank_account_id = None
+        if account_number and len(account_number) > 4:
+            iban_number = _pretty_iban(account_number)
+            bank_account_ids = self.env['res.partner.bank'].search(
```
Comment (pedrobaeza) [inline account_payment_return_import/models/account_payment_return_import.py]: Remove the ids from the variable name. It's a recordset

**Comment by carlosdauden** [comment]
Changes done.

**Comment by pedrobaeza** [comment]
Superseeded by https://github.com/OCA/account-payment/pull/78

**Pull request closed without merging**